src/main/java/ch/njol/skript/doc/HTMLGenerator.java

```diff
@@ -855,14 +852,9 @@ private String generateFunction(String descTemp, JavaFunction<?> info) {
 		// Assume element.pattern generate
 		for (String data : toGen) {
 			String[] split = data.split(" ");
-			String pattern = readFile(new File(template + "/templates/" + split[1]));
+			String pattern = readFile(new File(templateDir + "/templates/" + split[1]));
 			String patterns = "";
-			Parameter<?>[] params = info.getParameters();
-			String[] types = new String[params.length];
-			for (int i = 0; i < types.length; i++) {
-				types[i] = params[i].toString();
-			}
-			String line = docName + "(" + Joiner.on(", ").join(types) + ")"; // Better not have nulls
+			String line = info.getSignature().toString(false); // Better not have nulls
```

Comment:
```suggestion
			String line = info.getSignature().toString(true); // Better not have nulls
```
I say the function returntype would be nice to have :)

Reply:
it would be nice to have. the reason I didn't include it here is because it goes into the "patterns" field of the docs, and you don't specify the return type when calling a function. if that isn't something we care about, i'm happy to add it tho

Reply:
hmm, good point.. maybe we can add optional brackets around the returntype syntax just for docs? that would give a hint to the user that it's not compulsory